### PR TITLE
upload attachment animation with loading and abort (DEV-7)

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
DEV-7 

upload animation with loading and abort

### CODE REVIEW

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels ,

A couple of things are missing from what I saw from your PR.

1. The background of loading (light grey: #F4F6FD)
2. The Delete the Image message dialog (#5 from the Breakdown)

image
image
image

Let me know if you have any questions. Thanks!

File: libs/expo/betterangels/src/lib/ui-components/Attachments/ImageComponent.tsx [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
+} from '@monorepo/expo/shared/ui-components';
+import { Dispatch, SetStateAction } from 'react';
 import { Image, View } from 'react-native';
 import { DELETE_NOTE_ATTACHMENT } from '../../apollo';
 import {
   DeleteNoteAttachmentMutation,
   DeleteNoteAttachmentMutationVariables,
 } from '../../apollo/graphql/__generated__/mutations.generated';
 
+interface IImage {
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/ui-components/Attachments/ImageComponent.tsx:16]: just curious, does this have to be defined in each component? There's no way to reuse it?

File: libs/expo/shared/ui-components/src/lib/ImagePicker/index.tsx [hunk 2]
```diff
@@ -57,32 +63,67 @@ export default function ImagePickerComponent(props: IImagePickerProps) {
...
+
+        const uploadPromises = newImages.map(async (image) => {
+          const asset = result.assets.find((a) => a.uri === image.uri);
+          if (!asset) return;
           const resizedPhoto = await resizeImage({ uri: asset.uri });
           const file = new ReactNativeFile({
             uri: resizedPhoto.uri,
             name: asset?.fileName || Date.now().toString(),
-            type: asset.mimeType || 'changeme',
+            type: asset?.mimeType || 'changeme',
```
Comment (mikefeldberg) [inline libs/expo/shared/ui-components/src/lib/ImagePicker/index.tsx:89]: what's `'changeme'`?

**Review by mikefeldberg — COMMENTED** [review]
a few questions. Also, it looks like this isn't implementing % progress.

**Comment by Davit-BetterAngels** [comment]
> Hey @Davit-BetterAngels ,
> 
> A couple of things are missing from what I saw from your PR.
> 
> 1. The background of loading (light grey: #F4F6FD)
> 2. The Delete the Image message dialog ([Replace buildng Node & Python with nikolaik/python-nodejs docker image #5](https://github.com/BetterAngelsLA/monorepo/pull/5) from the Breakdown)
> 
> image image image
> 
> Let me know if you have any questions. Thanks!

1. As we don't have a progress the grey will not be filled so the idea of having grey background is losing the point. That is why I left blank to show loading indicator. 
2. will be done

File: libs/expo/betterangels/src/lib/ui-components/Attachments/ImageComponent.tsx (hunk 1 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/Attachments/ImageComponent.tsx:17]: if we create types file and import from there.

File: libs/expo/shared/ui-components/src/lib/ImagePicker/index.tsx (hunk 2 above)
Comment (Davit-BetterAngels) [reply libs/expo/shared/ui-components/src/lib/ImagePicker/index.tsx:89]: left that for me to find most usable and forgot. changed to imag jpeg

**Comment by Davit-BetterAngels** [comment]
> a few questions. Also, it looks like this isn't implementing % progress.

yes there is no way for now to get the progress with apollo. 

with custom fetch backend gives an error "dict has not 'name '"

so we have to come back to progress later

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels,

> 1. As we don't have a progress the grey will not be filled so the idea of having grey background is losing the point. That is why I left blank to show loading indicator.

Thanks for the comment, @Davit-BetterAngels. I remember the conversation in the standup/parking lot.
So here is my proposal then. Can we adjust the spinner to [this one](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=14593-62334&t=1QjJ7VYAuVx7SQTj-4) then? The animation is already there so I think you can just copy and paste the code. 


> 2\. will be done

Cool thanks! And to address what you mentioned in the standup/parking lot this morning (8/7/24), I added [the component](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=14593-62173&t=1QjJ7VYAuVx7SQTj-4) in the Components page in Figma. Please adjust the sizes accordingly. Thanks!

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels ,

I see the difference of padding between Delete Attachment and Delete Interaction.
If we can be consistent on 16px for mobile, that'd be great.
Other than that, everything else looks amazing! 👍 😄 

image
image

File: libs/expo/betterangels/src/lib/ui-components/Attachments/ImageComponent.tsx (hunk 1 above)
Comment (mikefeldberg) [reply libs/expo/betterangels/src/lib/ui-components/Attachments/ImageComponent.tsx:17]: I'm not as familiar with fe convention, so I'm not sure if/when we'd want to do that. Do you think it makes sense to have a types file now, or can we hold off on that? May be a good idea to create a ticket for it just so it's in the backlog and we know it's something we eventually want to do

**Comment by mikefeldberg** [comment]
> > a few questions. Also, it looks like this isn't implementing % progress.
> 
> yes there is no way for now to get the progress with apollo.
> 
> with custom fetch backend gives an error "dict has not 'name '"
> 
> so we have to come back to progress later

In that case, can you please
1. update the PR description to note that this PR only partially satisfies the ticket, including which AC were satisfied and which will be done later
2. create a ticket to cover the AC not included in this PR
3. include a link to that ticket in the PR description
4. include a link to that ticket in DEV-7

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels 

I'm not sure this is done yet but I was curious and checked. The animation looks great! However, I see something else now..

When I tapped the Photo Album icon, the camera icon next to it also got pressed/active at the same time as well. 
Please check. Thanks!

image

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels 

It looks good! Thanks for working on this!

**Review by shiozakiyuko — APPROVED** [review]

**Pull request merged**